File: Week_02/429/n_ary_tree_level_order_traversal.py
from typing import List


class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children


class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        res = []

        def level_order(node: Node, level: int):
            if node is not None:
                if len(res) == level:
                    res.append([node.val])
                else:
                    res[level].append(node.val)
                for c in node.children:
                    level_order(c, level + 1)

        level_order(root, 0)
        return res

File: Week_02/429/n_ary_tree_level_order_traversal.go
package main

type Node struct {
	Val      int
	Children []*Node
}

func levelOrder(root *Node) [][]int {
	res := make([][]int, 0)

	var levelOrder func(node *Node, level int)
	levelOrder = func(node *Node, level int) {
		if node != nil {
			if len(res) == level {
				res = append(res, []int{node.Val})
			} else {
				res[level] = append(res[level], node.Val)
			}
			for i := 0; i < len(node.Children); i++ {
				levelOrder(node.Children[i], level+1)
			}
		}
	}
	levelOrder(root, 0)
	return res
}
